### ISSUE TITLE
iOS 13 Safari is load time is Slow

### ISSUE THREAD

**Issue by justinbarry** [AugurProject/augur#3869]
Problem
===

non-private mode safari on ios 13 is super slow to load.

![image](https://user-images.githubusercontent.com/328965/65757550-3940a980-e0cc-11e9-8ac9-9d01bc8f6405.png)

Further complications
===

iOS 12 seems to work fine:
![image](https://user-images.githubusercontent.com/328965/65757521-2b8b2400-e0cc-11e9-8819-ac966336021e.png)

As does iOS 13 private mode:
![image](https://user-images.githubusercontent.com/328965/65757915-e9aead80-e0cc-11e9-9f0d-03f5bce549d4.png)

**Comment by pgebheim** [comment]
If you load up the profiler do you see anything at all interesting?

**Comment by robert-e-davidson3** [comment]
I did some investigating:

-----

On iOS, our app is fast to load in Private browsing mode but slow to load in Regular mode. The difference is about an order of magnitude. Having done some testing and research, I believe the cause is a combination of these two factors: 1) Private mode on iOS does not store the databases to disk and 2) our database, IndexedDB, is slow on iOS.

I did some googling and found articles claiming that inserts to IndexedDB on iOS can take multiple seconds(!). This is orders of magnitude slower than other platforms.

Looking at the Storage tab in Safari's debug mode, I found that the Regular mode uses IndexedDB but Private mode does not. This suggests that they use an in-memory database only for Private mode. This would virtually eliminate the time required to write to the database.

At first glance it's odd that, excepting iOS, writing to disk is as fast as writing to memory. However, blocking on network I/O should take up almost all of the load time. The remainder is within the margin of error except on iOS.

Note that this is still speculative. A definitive test would involve measuring write time to IndexedDB on iOS Regular vs Private modes. This should be added to the existing Augur app if possible, to eliminate variables like IndexedDB size and the numbers of writes, databases, and indexes. If write times are very large in Regular mode but minute in Private mode then we know that the problem is with iOS's IndexedDB implementation. Otherwise the problem lies elsewhere.

A potential solution is to use an in-memory database for fast load then leisurely replicate it onto disk. (credit: Paul)

tl;dr: It's probably a problem with iOS. We can run a definitive test by adding debugging info to our app on our next deploy.

**Comment by justinbarry** [comment]
Using https://augurproject.github.io/pouchdb-perf/ we have confirmed the cost stems from indexedDB db and index creation. 

questions here: 
====

1. Will reducing the number of active dbs/indexes speed up mobile safari? (Yes)
2. Is it a similar result for other browsers?

To research:
====

1. A replication scheme similar to the one stated above that only builds indexes in memory.

Ruled out:
====

1.  The Webworker does not seem to be source of the slowness.

**Comment by justinbarry** [comment]
Filed: https://github.com/pouchdb/pouchdb/issues/7928

Hopefully they can brining it to the attention of interested parties.... now to find a short term fix.....

**Comment by joeykrug** [comment]
1) if I navigate to another app, it seems to get messed up
2) debug may be enabled which makes it slower on our site 
3) each subsequent performance test is a lot slower on https://augurproject.github.io/pouchdb-perf/. This isn’t true for dexie. Idk why

**Comment by joeykrug** [comment]
switched to dexie
